### PR TITLE
working on upload images to AWS bucket

### PR DESCRIPTION
**Pull request by dilippoudel**
I am still confusion with aws id/credential variable, no test file so far :-D

### CODE REVIEW

File: src/controllers/uploader.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  next: NextFunction
+) => {
+  try {
+    const s3 = new aws.S3()
+    const fileName = req.body.fileName
+    const fileType = req.body.fileType
+    const s3Params = {
+      Bucket: S3_BUCKET,
+      Key: fileName,
+      Expires: 500,
```
Comment (Wyfy0107) [inline src/controllers/uploader.ts]: Do you know the unit here? Is it minutes?

File: src/controllers/uploader.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      ContentType: fileType,
+      ACL: 'public-read',
+    }
+    s3.getSignedUrl('putObject', s3Params, (err, data) => {
+      if (err) {
+        console.log(err)
```
Comment (Wyfy0107) [inline src/controllers/uploader.ts]: Remove

File: src/controllers/uploader.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        res.json({ success: false, error: err })
```
Comment (Wyfy0107) [inline src/controllers/uploader.ts]: Use the predefined api error that you imported

File: src/controllers/uploader.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      }
+      const returnData = {
+        signedRequest: data,
+        url: `https://${S3_BUCKET}.s3.amazonaws.com/${fileName}`,
+      }
+      res.json({ success: true, data: { returnData } })
+    })
+  } catch (e) {
+    console.log(e)
```
Comment (Wyfy0107) [inline src/controllers/uploader.ts]: Here also

File: src/routers/employer.ts [hunk 5]
```diff
@@ -17,5 +18,5 @@ router.post('/jobs', tokenVerify, createJobPost)
 router.patch('/jobs/:id', tokenVerify, updateJobPost)
 router.patch('/', tokenVerify, updateEmployer)
 router.delete('/jobs/:id', tokenVerify, deleteJobPostbyId)
-
+router.post('/upload/profile', tokenVerify, uploadImages)
```
Comment (Wyfy0107) [inline src/routers/employer.ts]: If both `jobseeker` and `employer` can use this route. Perhaps you can rename the login router as `commonRouter` where both user types can use it and put `uploadImages` controller in there.

File: src/controllers/uploader.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+      ACL: 'public-read',
+    }
+    s3.getSignedUrl('putObject', s3Params, (err, data) => {
+      if (err) {
+        console.log(err)
+        res.json({ success: false, error: err })
+      }
+      const returnData = {
+        signedRequest: data,
+        url: `https://${S3_BUCKET}.s3.amazonaws.com/${fileName}`,
```
Comment (Wyfy0107) [inline src/controllers/uploader.ts:38]: Don't we need to save this `url` inside the database as well for easier access next time use login?

**Review by Wyfy0107 — CHANGES_REQUESTED** [review]

File: src/controllers/uploader.ts (hunk 1 above)
Comment (dilippoudel) [reply src/controllers/uploader.ts]: this is following from article, i really don't know :??

Comment (Wyfy0107) [reply src/controllers/uploader.ts]: Try to read the docs: https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#getSignedUrl-property
This `signedUrl` will expire in 500 seconds

File: src/controllers/uploader.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        return new InternalServerError(err.message)
+      }
+      const returnData = {
+        signedRequest: data,
+        url: `https://${S3_BUCKET}.s3.amazonaws.com/${fileName}`,
+      }
+      res.json({ success: true, data: { returnData } })
+    })
+  } catch (e) {
+    new InternalServerError(e)
```
Comment (Wyfy0107) [inline src/controllers/uploader.ts]: `next(new InternalServerError(e))`

**Review by Wyfy0107 — CHANGES_REQUESTED** [review]
Can you remove the `image.png` file

**Comment by Wyfy0107** [comment]
@dilippoudel Can you update the pr so we can merge it?

File: src/controllers/uploader.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+import {
+  NotFoundError,
+  UnauthorizedError,
+  InternalServerError,
+  BadRequestError,
+} from '../helpers/apiError'
+aws.config.update({
+  region: 'us-east-1',
+  accessKeyId: process.env.AWSAccessKeyId,
+  secretAccessKey: process.env.AWSSecretKey,
```
Comment (Wyfy0107) [inline src/controllers/uploader.ts]: Can you rename the env variables also:
`AWS_ACCESS_KEY`
`AWS_SECRET_KEY`

**Review by Wyfy0107 — CHANGES_REQUESTED** [review]

File: src/controllers/uploader.ts (hunk 7 above)
Comment (dilippoudel) [reply src/controllers/uploader.ts]: it doesn't accept the parameter only e. it says typescript error `Argument of type 'Error' is not assignable to parameter of type 'string'.`
so it should be e.message

File: src/controllers/uploader.ts (hunk 8 above)
Comment (dilippoudel) [reply src/controllers/uploader.ts]: Done

File: src/controllers/uploader.ts (hunk 7 above)
Comment (Wyfy0107) [reply src/controllers/uploader.ts]: then you don't need to pass anything in there: `next(new InternalServerError())`

File: src/controllers/uploader.ts [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+        return next(new InternalServerError())
+      }
+      const returnData = {
+        signedRequest: data,
+        url: `https://${S3_BUCKET}.s3.amazonaws.com/${fileName}`,
+      }
+      res.json({ success: true, data: { returnData } })
+    })
+  } catch (e) {
+    next(new InternalServerError(e))
```
Comment (Wyfy0107) [inline src/controllers/uploader.ts]: you forgot this line

**Review by Wyfy0107 — CHANGES_REQUESTED** [review]

File: src/controllers/uploader.ts (hunk 9 above)
Comment (dilippoudel) [reply src/controllers/uploader.ts]: done

**Review by Wyfy0107 — APPROVED** [review]

**Pull request merged**